pkg/sensors/tracing/kprobe_test.go

```diff
@@ -4771,68 +4776,75 @@ spec:
 `
 	}
 
-	var doneWG, readyWG sync.WaitGroup
-	defer doneWG.Wait()
+	if err := observer.InitDataCache(1024); err != nil {
+		t.Fatalf("observertesthelper.InitDataCache: %s", err)
+	}
 
-	ctx, cancel := context.WithTimeout(context.Background(), tus.Conf().CmdWaitTime)
-	defer cancel()
+	option.Config.HubbleLib = tus.Conf().TetragonLib
+	tus.LoadInitialSensor(t)
+	tus.LoadSensor(t, testsensor.GetTestSensor())
+	sm := tuo.GetTestSensorManager(t)
 
-	createCrdFile(t, hook)
-	obs, err := observertesthelper.GetDefaultObserverWithFile(t, ctx, testConfigFile, tus.Conf().TetragonLib)
+	tp, err := tracingpolicy.FromYAML(hook)
 	if err != nil {
-		t.Fatalf("GetDefaultObserverWithFile error: %s", err)
+		t.Fatalf("failed to decode yaml: %s", err)
 	}
-	observertesthelper.LoopEvents(ctx, t, &doneWG, &readyWG, obs)
-	readyWG.Wait()
 
-	server := "nc.openbsd"
-	cmdServer := exec.Command(server, "-unvlp", "9468", "-s", "127.0.0.1")
-	require.NoError(t, cmdServer.Start())
-	time.Sleep(1 * time.Second)
+	err = sm.Manager.AddTracingPolicy(ctx, tp)
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		sm.Manager.DeleteTracingPolicy(ctx, "datagram", "")
+	})
 
-	// Generate 5 datagrams
-	socket, err := net.Dial("udp", "127.0.0.1:9468")
-	if err != nil {
-		t.Fatalf("failed dialing socket: %s", err)
-	}
+	ops := func() {
+		server := "nc.openbsd"
+		cmdServer := exec.Command(server, "-unvlp", "9468", "-s", "127.0.0.1")
+		require.NoError(t, cmdServer.Start())
+		time.Sleep(1 * time.Second)
```

Comment:
is this sleep needed? any chance the dial call below would wait long enough to connect?